### PR TITLE
Don't "flatten" nested if expressions when formatting

### LINKED ISSUE #6939: Formatter: Nested if expression formatting
What I can tell from the Django diff is that Black uses newlines to separate all `if` and `else` branches if any `ExprIf` in the chain breaks:

```python
# Black
_check_for_duplicates(
    "bases",
    (
        base._meta.label_lower
        if hasattr(base, "_meta")
        else base.lower()
        if isinstance(base, str)
        else base
        for base in self.bases
    ),
)
        
clone._iterable_class = (
    NamedValuesListIterable
    if named
    else FlatValuesListIterable
    if flat
    else ValuesListIterable
)

# Ruff

_check_for_duplicates(
    "bases",
    (
        base._meta.label_lower
        if hasattr(base, "_meta")
        else base.lower() if isinstance(base, str) else base
        for base in self.bases
    ),
)

clone._iterable_class = (
    NamedValuesListIterable
    if named
    else FlatValuesListIterable if flat else ValuesListIterable
)
```

Notice how ruff "flattens" the nested else clauses. 

The fix for this is likely to special case the formatting of the `else` clause if it contains a nested `ExprIf` and pass a custom layout that doesn't introduce a new `group` (it should reuse the group of the parent to ensure that either both of them expand). This is similar to the `BoolOp` formatting

### PR DESCRIPTION
**Pull request by LaBatata101**
## Summary

Closes #6939

## Test Plan

`cargo test`

### CODE REVIEW

**Review by charliermarsh — APPROVED** [review]
LGTM -- added some additional tests and verified against Black.

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@conditional_expression.py.snap [hunk 1]
```diff
@@ -136,6 +136,15 @@ def something():
...
  )
  
+@@ -86,5 +78,7 @@
+     clone._iterable_class = (
+         NamedValuesListIterable
+         if named
+-        else FlatValuesListIterable if flat else ValuesListIterable
++        else FlatValuesListIterable
++        if flat
++        else ValuesListIterable
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/tests/snapshots/black_compatibility@conditional_expression.py.snap:146]: This seems to be a regression

**Comment by MichaReiser** [comment]
This PR regresses formatting of 

```python
# Reduce Model instances to their primary key values
row_data = tuple(
    d._get_pk_val()
    if hasattr(d, "_get_pk_val")
    # Prevent "unhashable type: list" errors later on.
    else tuple(d)
    if isinstance(d, list)
    else d
    for d in row_data
)
if row_data and None not in row_data:
    pass
```
Black collapses the second `else` as it seems because of the comment. 

```python
# Reduce Model instances to their primary key values
row_data = tuple(
    d._get_pk_val() if hasattr(d, "_get_pk_val")
    # Prevent "unhashable type: list" errors later on.
    else tuple(d) if isinstance(d, list) else d
    for d in row_data
)
if row_data and None not in row_data:
    pass
```

without the comment

```python
# Reduce Model instances to their primary key values
row_data = tuple(
    d._get_pk_val()
    if hasattr(d, "_get_pk_val")
    else tuple(d)
    if isinstance(d, list)
    else d
    for d in row_data
)
if row_data and None not in row_data:
    pass
```

I prefer our formatting because there's a smaller difference between with/without comment.

**Comment by MichaReiser** [comment]
Thank you!

**Pull request merged**